pygmt/tests/test_io_gmtread.py

```diff
@@ -0,0 +1,46 @@
+"""
+Test the gmtread function.
+"""
+
+import importlib
+
+import numpy as np
+import pytest
+import xarray as xr
+from pygmt import gmtread, which
+
+_HAS_NETCDF4 = bool(importlib.util.find_spec("netCDF4"))
+
+
+@pytest.mark.skipif(not _HAS_NETCDF4, reason="netCDF4 is not installed.")
+def test_io_gmtread_grid():
+    """
+    Test that reading a grid returns an xr.DataArray and the grid is the same as the one
+    loaded via xarray.load_dataarray.
+    """
+    grid = gmtread("@static_earth_relief.nc", kind="grid")
+    assert isinstance(grid, xr.DataArray)
+    expected_grid = xr.load_dataarray(which("@static_earth_relief.nc", download="a"))
+    assert np.allclose(grid, expected_grid)
```

Comment:
Also should have a similar test for `kind="image"`, comparing against `rioxarray.open_rasterio`?

Reply:
Done in a6c4ee70e87af2a9ecaeab5e8c352cbab819896c.

When I tried to add a test for reading datasets, I realized that the DataFrame returned by the `load_sample_data` is not ideal:
```
In [1]: from pygmt.datasets import load_sample_data

In [2]: data = load_sample_data("hotspots")

In [3]: data.dtypes
Out[3]: 
longitude      float64
latitude       float64
symbol_size    float64
place_name      object
dtype: object
```
The last column `place_name` should be string dtype, rather than `object`. We also have similar issues for other sample datasets. 

We have three options:

1. Do nothing and keep them unchanged
2. Fix and use appropriate dtypes
3. Use the new `gmtread` function instead of `pd.read_csv` in `_load_xxx` functions.

I'm inclined to option 3.

Reply:
> 3\. Use the new `gmtread` function instead of `pd.read_csv` in `_load_xxx` functions.
> 
> I'm inclined to option 3.

Agree with this. We should also add dtype related checks for the tabular dataset tests in `pygmt/tests/test_datasets_samples.py`.